src/emulate.c

```diff
@@ -957,25 +962,48 @@ static void match_pattern(riscv_t *rv, block_t *block)
         case rv_insn_lw:
             COMBINE_MEM_OPS(1);
             break;
-        case rv_insn_lui:
-            next_ir = ir + 1;
-            if (next_ir->opcode == rv_insn_addi && ir->rd == next_ir->rs1) {
-                /* The destination register of the LUI instruction is the
-                 * same as the source register 1 of the next instruction ADDI.
-                 */
-                ir->opcode = rv_insn_fuse5;
-                ir->rs1 = next_ir->rd;
-                ir->imm2 = next_ir->imm;
-                ir->impl = dispatch_table[ir->opcode];
-                ir->tailcall = next_ir->tailcall;
-            }
-            break;
             /* TODO: mixture of SW and LW */
             /* TODO: reorder insturction to match pattern */
         }
     }
 }
 
+
+typedef struct {
+    bool is_constant[N_RV_REGS];
```

Comment:
Does it make sense to check x0, which is always zero?

Reply:
Added